src/tpm.rs

```diff
@@ -33,6 +33,73 @@ ftpm_initialize.py
 Following are function from tpm_initialize.py program
 *****************************************************************/
 
+/*
+ * Return: Result wrap named temp file name or error
+ *
+ * Make a temp file and return its file path.
+ */
+fn create_and_get_temp_file_path() -> Result<String, Box<String>> {
+    match NamedTempFile::new() {
+        Ok(tmp_file) => match tmp_file.path().to_str() {
+            Some(s) => Ok(s.to_string()),
+            None => emsg(
+                format!("Failed to get temp file path string. Error")
+                    .as_str(),
+                None::<String>,
+            ),
+        },
+        Err(e) => emsg(
+            format!("Failed to create a  tempfile. Error {}.", e).as_str(),
+            Some(e),
+        ),
+    }
+}
+
+/*
+ * Return: Result unwrap execution result
+ *
+ * Execute TPM command and retrip the public ek key and save to
+ * tpmdata.json file. Same implementation as the original python
+ * version with same error hanlding procedures.
+ */
+fn get_pub_ek() -> Result<(), Box<String>> {
+    let owner_password = get_tpm_metadata_content("owner_pw")?;
+    let tmp_path = create_and_get_temp_file_path()?;
+
+    let (return_output, return_code, ek) = run(
+        format!("getpuek -pwdo {} -ok {}", owner_password, tmp_path),
```

Comment:
As said in https://github.com/keylime/rust-keylime/pull/39#pullrequestreview-203984790 `tmp_path` - this could be risky. 

Does this have some randomness in the name generation so its not easily guessed, a bit like bash's mktemp

 ```
 mktemp 
/tmp/tmp.8kvXUx3Pxv
```